### PR TITLE
Add more model APIs

### PR DESCRIPTION
**Pull request by chapulina**
Partially addresses #325

Added a new tutorial / guide which lists all of `gazebo::physics::Model` functions and how they map to existing Ignition Gazebo functionality.

Also added several functions to `Model.hh` and `Util.hh`.

Finally, I made more of the model's components be displayed on the component inspector, related to #158.

### CODE REVIEW

File: include/ignition/gazebo/Model.hh [hunk 1]
```diff
@@ -95,11 +98,41 @@ namespace ignition
...
 
+      /// \brief Get the ID of the entity which is the immediate parent of this
+      /// model.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return Parent entity ID.
+      public: gazebo::Entity Parent(const EntityComponentManager &_ecm) const;
+
+      /// \brief Get whether this model is static.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return True if self-colliding.
```
Comment (JShep1) [inline include/ignition/gazebo/Model.hh:109]:
```suggestion
      /// \return True if static.
```

File: include/ignition/gazebo/Util.hh [hunk 2]
```diff
@@ -51,6 +51,50 @@ namespace ignition
...
+    /// `gazebo::components::Model::typeId == entityTypeId(entity, ecm)`
+    ///
+    /// In case the entity isn't of any known type, this will return
+    /// `ignition::gazebo::kComponentTypeIdInvalid`.
+    ///
+    /// In case the entity has more than one type, only one of them will be
+    /// returned. This is not standard usage.
+    ///
+    /// \return ID of entity's type-defining components.
+    ComponentTypeId IGNITION_GAZEBO_VISIBLE entityTypeId(const Entity &_entity,
```
Comment (JShep1) [inline include/ignition/gazebo/Util.hh:72]: Do we need `param[in]`s for this and below functions in this file?

File: include/ignition/gazebo/components/LinearVelocityCmd.hh [hunk 3]
```diff
@@ -32,15 +32,17 @@ namespace gazebo
 inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
 namespace components
 {
-  /// \brief A component type that contains linear velocity of an entity
-  /// represented by ignition::math::Vector3d.
+  /// \brief A component type that contains commandad linear velocity of an
```
Comment (JShep1) [inline include/ignition/gazebo/components/LinearVelocityCmd.hh:35]: I'm not sure what `commandad` is supposed to mean here, same for below comment

File: src/systems/physics/Physics.cc [hunk 4]
```diff
@@ -1409,6 +1409,15 @@ void PhysicsPrivate::UpdatePhysics(EntityComponentManager &_ecm)
             this->entityWorldVelocityCommandMap);
         if (!worldAngularVelFeature)
         {
+          static bool informed{false};
```
Comment (JShep1) [inline src/systems/physics/Physics.cc:1412]: What is the purpose of `informed` here? Just so the debug message is only printed once?

File: tutorials/migration_model_api.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+# Migration from Gazebo-classic: Model API
+
+When migrating plugins from Gazebo-classic to Ignition Gazebo, developers will
+notice that the C++ APIs for both simulators are quite different. Be sure to
+check the [plugin migration tutorial](migrationplugins.html) to get a high-level
+view of the architecture differences before using this guide.
+
+This tutorial is meant to serve as a reference guide for developers migrating
+functions from the
+[gazebo::phyiscs::Model](http://osrf-distributions.s3.amazonaws.com/gazebo/api/11.0.0/classgazebo_1_1physics_1_1Model.html)
```
Comment (JShep1) [inline tutorials/migration_model_api.md:12]:
```suggestion
[gazebo::physics::Model](http://osrf-distributions.s3.amazonaws.com/gazebo/api/11.0.0/classgazebo_1_1physics_1_1Model.html)
```

**Review by JShep1 — CHANGES_REQUESTED** [review]
LGTM, I left a couple small nits and a question, but pending that and CI, this should be good to go.

File: include/ignition/gazebo/components/LinearVelocityCmd.hh (hunk 3 above)
Comment (chapulina) [reply include/ignition/gazebo/components/LinearVelocityCmd.hh:35]: It's when your dad commands you to do something :upside_down_face:

File: src/systems/physics/Physics.cc (hunk 4 above)
Comment (chapulina) [reply src/systems/physics/Physics.cc:1412]: Yup, this is the pattern we've been using throughout this file to avoid spamming the user.

**Review by JShep1 — APPROVED** [review]
Pending CI this is good to go!

**Pull request merged**